figure_generation/01_figure_generation.py

```diff
@@ -0,0 +1,364 @@
+    gen_model_performance_all_df
+    >> ply.call(".append", gen_model_performance_df)
+    >> ply.query("data_source=='abstract'")
+    >> ply.call(".append", beginning_point_df)
+    >> ply.define(lf_num=ply.expressions.if_else("lf_num=='All'", 99, "lf_num"))
+    >> ply.define(lf_num="lf_num.astype(int)")
+    >> ply.define(
+        lf_num=ply.expressions.case_when(
+            {
+                "lf_num==1": "lf_num",
+                "lf_num==6": 33,
+                "lf_num==11": 65,
+                "lf_num==16": 97,
+                True: "lf_num",
+            }
+        )
+    )
+)
+gen_model_performance_all_df.lf_num = pd.Categorical(
+    gen_model_performance_all_df.lf_num.tolist(), categories=[0, 1, 33, 65, 97, 99]
+)
+gen_model_performance_all_df >> ply.slice_rows(10)
+
+g = (
+    p9.ggplot(
+        gen_model_performance_all_df
+        >> ply.query("label_source==prediction_label|label_source=='ALL'")
+        >> ply.query("model=='test'")
+    )
+    + p9.aes(
+        x="lf_num",
+        y="auroc_mean",
+        ymin="auroc_lower_ci",
+        ymax="auroc_upper_ci",
+        group="label_source",
+        color="label_source",
+    )
+    + p9.geom_point()
+    + p9.geom_line()
+    + p9.geom_errorbar()
+    + p9.facet_wrap("~ prediction_label")
+    + p9.theme_bw()
+    + p9.theme(figure_size=(8, 6))
+    + p9.scale_color_manual(values=color_map)
+    + p9.labs(
+        title="Generative Model Performance using All Label Functions (Test Set/AUROC)",
+        color="Relation (LF) Source",
+        x="Number of Label Functions Added",
+        y="AUROC",
+    )
+)
+g.save("output/figure_four.svg")
+g.save("output/figure_four.png", dpi=300)
+print(g)
+
+g = (
+    p9.ggplot(
+        gen_model_performance_all_df
+        >> ply.query("label_source==prediction_label|label_source=='ALL'")
+        >> ply.query("model=='test'")
+    )
+    + p9.aes(
+        x="lf_num",
+        y="aupr_mean",
+        ymin="aupr_lower_ci",
+        ymax="aupr_upper_ci",
+        group="label_source",
+        color="label_source",
+    )
+    + p9.geom_point()
+    + p9.geom_line()
+    + p9.geom_errorbar()
+    + p9.facet_wrap("~ prediction_label")
+    + p9.theme_bw()
+    + p9.theme(figure_size=(8, 6))
+    + p9.scale_color_manual(values=color_map)
+    + p9.labs(
+        title="Generative Model Performance using All Label Functions (Test Set/AUPR)",
+        color="Relation (LF) Source",
+        x="Number of Label Functions Added",
+        y="AUPR",
+    )
+)
+g.save("output/figure_five.svg")
+g.save("output/figure_five.png", dpi=300)
+print(g)
+
+# # Discriminator Model Figures
+
+# ## Figure 5 - Discriminator Model vs Generative Model
+
+disc_performance_df = pd.read_csv(
+    Path("../discriminative_model_training/output/all_total_lf_performance.tsv"),
+    sep="\t",
+)
+disc_performance_df.lf_num = pd.Categorical(
+    disc_performance_df.lf_num.tolist(), categories=["0", "1", "6", "11", "16", "All"]
```

Comment:
What are these categories? Could they ever change based on the data, or will they always be the same?

Reply:
>What are these categories? Could they ever change based on the data, or will they always be the same?

These categories are the amount of label functions I'm sampling to train the generative model (to generate labels for the discriminator model) for these experiments. They won't change as its always fixed.